modules/process/manager_stacktraces.go

```diff
@@ -0,0 +1,256 @@
+		}
+	}
+	if runInLock != nil {
+		runInLock()
+	}
+	pm.mutex.Unlock()
+
+	sort.Slice(processes, func(i, j int) bool {
+		left, right := processes[i], processes[j]
+
+		return left.Start.Before(right.Start)
+	})
+
+	return processes
+}
+
+// ProcessStacktraces gets the processes and stacktraces in a thread safe manner
+func (pm *Manager) ProcessStacktraces(flat, onlyRequests bool) ([]*Process, int64, error) {
+	var stacks *profile.Profile
+	var err error
+	processes := pm.Processes(false, false, func() {
+		reader, writer := io.Pipe()
+		defer reader.Close()
+		go func() {
+			err := pprof.Lookup("goroutine").WriteTo(writer, 0)
+			_ = writer.CloseWithError(err)
+		}()
+		stacks, err = profile.Parse(reader)
+		if err != nil {
+			return
+		}
+	})
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// We cannot use the process pidmaps here because we have released the mutex ...
+	pidMap := map[IDType]*Process{}
+	processStacks := make([]*Process, 0, len(processes))
+	for _, process := range processes {
+		pStack := &Process{
+			PID:         process.PID,
+			ParentPID:   process.ParentPID,
+			Description: process.Description,
+			Start:       process.Start,
+			Type:        process.Type,
+		}
+
+		pidMap[process.PID] = pStack
+		if flat {
+			processStacks = append(processStacks, pStack)
+		} else if parent, ok := pidMap[process.ParentPID]; ok {
+			parent.Children = append(parent.Children, pStack)
+		} else {
+			processStacks = append(processStacks, pStack)
+		}
+	}
+
+	goroutineCount := int64(0)
+
+	// Now walk through the "Sample" slice in the goroutines stack
+	for _, sample := range stacks.Sample {
+		stack := &Stack{}
+
+		// Add the labels
+		for name, value := range sample.Label {
+			if name == DescriptionPProfLabel || name == PIDPProfLabel || (!flat && name == PPIDPProfLabel) || name == ProcessTypePProfLabel {
+				continue
+			}
+			if len(value) != 1 {
+				// Unexpected...
+				return nil, 0, fmt.Errorf("label: %s in goroutine stack with unexpected number of values: %v", name, value)
+			}
+
+			stack.Labels = append(stack.Labels, &Label{Name: name, Value: value[0]})
+		}
+
+		stack.Count = sample.Value[0]
+		goroutineCount += stack.Count
+
+		// Now get the processStack for this goroutine sample
+		var processStack *Process
+		if pidvalue, ok := sample.Label[PIDPProfLabel]; ok && len(pidvalue) == 1 {
+			pid := IDType(pidvalue[0])
+			processStack, ok = pidMap[pid]
+			if !ok && pid != "" {
+				ppid := IDType("")
+				if value, ok := sample.Label[PPIDPProfLabel]; ok && len(value) == 1 {
+					ppid = IDType(value[0])
+				}
+				description := "(dead process)"
+				if value, ok := sample.Label[DescriptionPProfLabel]; ok && len(value) == 1 {
+					description = value[0] + " " + description
+				}
+				ptype := "code"
+				if value, ok := sample.Label[ProcessTypePProfLabel]; ok && len(value) == 1 {
+					stack.Labels = append(stack.Labels, &Label{Name: ProcessTypePProfLabel, Value: value[0]})
```

Comment:
Should ptype be changed here? Or is it hardcoded to `code`?

Reply:
I think we want this to be hardcoded. I've chosen code because it will get the icon `<>` like the unlabelled processes.

I don't think we should have this keep the old type.